### PR TITLE
Fix CI timeout: Extract Frame tests to AbstractFrameTestBase

### LINKED ISSUE #111: BLOCKER: Build fail on long unit test
Run unit tests
The action 'Run unit tests' has timed out after 10 minutes.

Test opening Frame must be integration, and Frame must be correctly released (AbstractFrameTest)

### PR DESCRIPTION
**Pull request by bedaHovorka**
## Summary

Creates AbstractFrameTestBase as a base class for tests that open Frame GUI components, fixing the CI timeout issue where Frame instances were never disposed.

## Changes

- ✅ Created `AbstractFrameTestBase.kt` with Frame lifecycle management
- ✅ Created `MainEditModeTest.kt` extending AbstractFrameTestBase
- ✅ Extracted 3 edit-mode tests from `MainArgumentParsingTest.kt`
- ✅ Tagged as `@Tag("integration-test")` to run separately
- ✅ Added 10-second timeouts to prevent indefinite hangs
- ✅ Proper Frame disposal via Koin DI access
- ✅ Headless environment detection

## Fixes

Closes #111

The CI timeout was caused by tests calling `main(arrayOf("edit"))` which created Frame instances that were never disposed, causing the build to hang for 10 minutes.

## Test Plan

- [ ] Run `./gradlew test` to verify unit tests pass (excludes Frame tests)
- [ ] Run `./gradlew integrationTest` to verify Frame tests pass
- [ ] Verify tests complete within timeout limits
- [ ] Confirm no resource leaks or hanging processes

Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Comment by bedaHovorka** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/bedaHovorka/interlockSim/pull/112#pullrequestreview-3669416210)

File: src/test/kotlin/cz/vutbr/fit/interlockSim/gui/MainEditModeTest.kt [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+			if (koin != null) {
+				// Use getOrNull to avoid creating Frame if it wasn't instantiated yet
+				val frame = koin.getOrNull<Frame>()
+				if (frame != null) {
+					frames.add(frame)
+				}
+			}
+		} catch (e: Exception) {
+			// Frame might not have been created, that's okay
+		}
```
[the remark this answers is not in the document]
Comment (bedaHovorka) [reply src/test/kotlin/cz/vutbr/fit/interlockSim/gui/MainEditModeTest.kt:101]: Logger.debug exception @copilot

**Comment by bedaHovorka** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/bedaHovorka/interlockSim/pull/112#pullrequestreview-3669618998)

**Comment by bedaHovorka** [comment]
> ## Pull Request Review: Fix CI timeout - Extract Frame tests to AbstractFrameTestBase
> ### Overview
> This PR addresses issue #111 by extracting Frame-related tests into a dedicated class with proper lifecycle management. The solution demonstrates excellent software engineering practices.
> 
> ### ✅ Strengths
> **1. Excellent Architecture & Design**
> 
> * AbstractFrameTestBase provides reusable infrastructure for all future Frame tests
> * Clean separation of concerns with proper inheritance hierarchy
> * Integration test tagging ensures proper test categorization
> 
> **2. Comprehensive Resource Management**
> 
> * Frame disposal on EDT via SwingUtilities.invokeAndWait (thread-safe)
> * Nullable property handling for JUnit lifecycle edge cases
> * Koin cleanup prevents DI container pollution
> 
> **3. Robust Error Handling**
> 
> * Headless environment detection with Assumptions.assumeFalse
> * Multiple exception types handled: HeadlessException, AWTError, generic exceptions
> * isGuiRelatedException() helper validates exception types
> 
> **4. Excellent Documentation**
> 
> * Comprehensive KDoc comments explaining design decisions
> * Clear references to GitHub issue [BLOCKER: Build fail on long unit test #111](https://github.com/bedaHovorka/interlockSim/issues/111)
> * Usage examples in AbstractFrameTestBase
> 
> ### ⚠️ Issues & Recommendations
> **1. CRITICAL: Koin Double Initialization Risk** 🔴
> 
> Location: MainEditModeTest.kt:67-76
> 
> Problem: MainEditModeTest extends AbstractFrameTestBase, which extends KoinTestBase. This means KoinTestBase.setUpKoin() starts Koin with testModule, then main() calls startKoin() with interlockSimModule, causing KoinAppAlreadyStartedException.
> 
> Solution: Add stopKoin() before calling main():
> 
> ```kotlin
> @BeforeEach
> override fun setUp() {
>     super.setUp()  // Headless check
>     stopKoin()  // Clean slate for main()
>     // ... rest
> }
> ```
> 
> Impact: HIGH - Will cause test failures
> 
> **2. MEDIUM: Frame Singleton May Cause Resource Leaks** 🟡
> 
> Location: InterlockSimModule.kt
> 
> Problem: Frame registered as singleton, but if exception occurs before Frame() completes, koin.getOrNull() returns null and Frame never disposed.
> 
> Recommendation: Use explicit Frame tracking via AWT Window.getWindows()
> 
> Impact: MEDIUM - May cause intermittent CI failures
> 
> **3. MINOR: Test Assertions Could Be More Specific** 🟡
> 
> Location: MainEditModeTest.kt lines 134-161
> 
> Tests verify mode selection indirectly. Consider verifying Frame creation directly.
> 
> **4. MINOR: Unused Import** 📝
> 
> assertk.assertions.contains imported but never used
> 
> ### 🎯 Summary
> **Overall Assessment:** ⚠️ **Requires Changes Before Merge**
> 
> Excellent architecture and documentation, but Critical Issue #1 (Koin double initialization) must be fixed first.
> 
> **Action Items:**
> 
> 1. 🔴 MUST FIX: Add stopKoin() in setUp() before calling main()
> 2. 🟡 SHOULD FIX: Use AWT Window tracking for Frame disposal
> 3. 🟢 NICE TO HAVE: Improve assertions, remove unused import
> 4. ✅ VERIFY: Run full test suite and confirm CI passes
> 
> Great work on the architecture! The base class will serve as excellent infrastructure for future GUI tests. 🚀

@claude : stop in setup?

**Comment by bedaHovorka** [comment]
> **Claude finished @bedaHovorka's task in 46s** —— [View job](https://github.com/bedaHovorka/interlockSim/actions/runs/21061768517)
> 
> ### Analysis Complete: The Review's Recommendation is INCORRECT ❌
> I've analyzed the code and the review's recommendation to add `stopKoin()` in `setUp()` is **incorrect**. Here's why:
> 
> #### Tasks
> * [x]  Read MainEditModeTest.kt implementation
> * [x]  Read AbstractFrameTestBase.kt
> * [x]  Read KoinTestBase to understand lifecycle
> * [x]  Read Main.kt to understand startKoin() call
> * [x]  Analyze the initialization flow
> * [x]  Provide recommendation
> 
> ### The Actual Flow
> **Current implementation (CORRECT):**
> 
> 1. **@beforeeach**: `KoinTestBase.setUpKoin()` starts Koin with `testModule` (line 45-48 in KoinTestBase.kt)
> 2. **Test execution**: `main(arrayOf("edit"))` is called
> 3. **Inside main()**: `startKoin()` is called **again** with `interlockSimModule` (Main.kt:131)
> 4. **Result**: Second `startKoin()` throws `KoinAppAlreadyStartedException` ❌
> 
> **What the review suggested (STILL WRONG):**
> 
> ```kotlin
> @BeforeEach
> override fun setUp() {
>     super.setUp()  // Calls KoinTestBase.setUpKoin() → starts Koin
>     stopKoin()     // Stop Koin
>     // But now super.setUp() already started Koin!
> }
> ```
> 
> This doesn't help because:
> 
> * `super.setUp()` calls `AbstractFrameTestBase.setUp()` which calls `KoinTestBase.setUpKoin()`
> * `KoinTestBase.setUpKoin()` starts Koin with testModule
> * Then `main()` tries to start Koin again → still fails!
> 
> ### The Real Problem
> **MainEditModeTest should NOT extend KoinTestBase at all!**
> 
> Looking at MainEditModeTest.kt:40-41, the comment already says this:
> 
> ```kotlin
> * - Does NOT extend KoinTestBase to avoid double Koin initialization
> ```
> 
> But then the code **does extend KoinTestBase** via the inheritance chain:
> 
> ```
> MainEditModeTest → AbstractFrameTestBase → KoinTestBase
> ```
> 
> ### The Correct Solution
> **Option 1: Break KoinTestBase inheritance** (Recommended)
> 
> ```kotlin
> // AbstractFrameTestBase.kt - Remove KoinTestBase
> abstract class AbstractFrameTestBase {  // Don't extend KoinTestBase
>     // ... rest of code
> }
> ```
> 
> This way:
> 
> * Tests that need Koin for testing can extend KoinTestBase directly
> * Tests that call `main()` (which starts its own Koin) won't have conflicts
> * MainEditModeTest gets a clean slate for `main()` to start Koin
> 
> **Option 2: Stop Koin BEFORE calling super.setUp()**
> 
> ```kotlin
> @BeforeEach
> override fun setUp() {
>     stopKoin()  // Stop any previous Koin (including from KoinTestBase)
>     super.setUp()  // Now checks headless
>     // main() will start fresh Koin
> }
> ```
> 
> But this is hacky and defeats the purpose of inheriting from KoinTestBase.
> 
> ### My Recommendation
> **Remove KoinTestBase from AbstractFrameTestBase's inheritance chain.**
> 
> The comment in MainEditModeTest.kt:40 already identified this requirement. AbstractFrameTestBase should be Koin-agnostic since:
> 
> 1. It's a base class for Frame lifecycle management, not DI
> 2. Some tests may need testModule (via KoinTestBase)
> 3. Some tests may need interlockSimModule (via main())
> 4. Each test should choose its DI strategy
> 
> Would you like me to create a PR to fix this by removing KoinTestBase inheritance from AbstractFrameTestBase?

@copilot option 2 in new pr please
